c/ovm.c

```diff
@@ -1358,23 +1357,23 @@ word vm(word *ob, word *arg) {
       R[ip[1]] = (immediatep(ob)) ? IFALSE : F(hdrsize(*ob)-1);
       NEXT(2); }
    op37: /* lraw lst type r (FIXME: alloc amount testing compiler pass not in place yet) */
-      A2 = prim_lraw(A0, fixval(A1));
+      A2 = prim_lraw(A0, immval(A1));
       NEXT(3);
    op38: { /* fx+ a b r o, types prechecked, signs ignored, assume fixnumbits+1 fits to machine word */
-      word res = fixval(A0) + fixval(A1);
+      word res = immval(A0) + immval(A1);
       word low = res & FMAX;
       A3 = BOOL(res & (1 << FBITS));
       A2 = F(low);
       NEXT(4); }
    op39: { /* fx* a b l h */
-      uint64_t res = ((uint64_t) ((uint64_t) fixval(R[*ip])) * ((uint64_t) fixval(A1)));
+      uint64_t res = (uint64_t)immval(R[*ip]) * (uint64_t)immval(A1);
```

Comment:
R[*ip]:s could later be replaced with A0:s for clarity.

Reply:
already working on it (-: